### DISCUSSION TITLE
폴더구조 변경 및 통일 건의

### DISCUSSION THREAD

**Discussion by ppskj178** [boostcampaitech2/object-detection-level2-cv-06#57] [ideas]
줌에서 다른분들 도와주다보니 폴더 구조가 통일 되지 않아 문제가 발생한 부분도 있고
초기에 임의로 구성하다보니 비효율적인 부분도 있어보여
효율적인 작업 파이프 라인을 위해
아래 이미지와 같은 구조로 폴더구조를 통일 하고자합니다.

직관적이라 생각해서 따로 코멘트는 안다는데 혹시 이해 안가시는 부분 있으면 댓글 부탁드립니다.
추가적으로 해당 구조로 변경시에 `mmconfig` 내에 #030_deformable_detr, cascade_swin_ms_crop 폴더 내용물을 분해해서 메인 config만 남길 생각입니다. 그리고 해당 이름으로 work dir에 경로가 생기기 때문에 폴더명까지 관리하는 비효율이 해결될 것으로 보입니다.

![image](https://user-images.githubusercontent.com/20790778/135826499-100170ad-e2ab-4cfb-adbe-b4c5a49925c2.png)

**Comment by jaehwan-AI** [comment]
하나의 모델에 대해 변경된 config 파일들이 흩어지는 것 같은 느낌이 드는데 저렇게 관리하면 어떤 이점이 있을까요?

**Reply by ppskj178** [reply]
저도 처음에 그생각으로 원래 폴더구조를 생각한건데
본문에 작성한 이유 말고도
1. mmdetection 폴더 구조와 최대한 유사해진다
2. 좋은 모델이나 데이터셋이 발견되면 해당 테스트셋팅을 매번 참조하게 될것 같은데 기존의 방식으론 이 경우 번거롭고 경로가 직관적이지 못해진다
3. 어차피 우리가 주로 볼것은 여러 config를 합친 final(?) config 파일이라 굳이 모두 모아놓을 필요가 없다.

입니다.

**Reply by jaehwan-AI** [reply]
관리나 네이밍 측면에서 위와 같이 구조를 통일하는 것이 좋을 것 같습니다.

**Reply by sne12345** [reply]
전 좋다고 생각합니다 !

**Reply by JiyouSeo** [reply]
분해한다는 말씀이 조금 이해가 힘든데, deformable_detr, cascade_swin_ms_crop이 있다고 할 때,
main config외에 다른 dataset, scheduler, runtime, model 이 config들을 어떤식으로 분해하고 싶으신 건가요?

**Reply by ppskj178** [reply]
> 분해한다는 말씀이 조금 이해가 힘든데, deformable_detr, cascade_swin_ms_crop이 있다고 할 때, main config외에 다른 dataset, scheduler, runtime, model 이 config들을 어떤식으로 분해하고 싶으신 건가요?

dataset, scheduler, runtime, model에 해당하는 config는 해당 폴더에 넣어두고
test폴더내에 main_config를 선언해서 
dataset, scheduler, runtime, model를 불러오는 방식입니다.

이후에 test 하다가 성능이 좋은것들이 발견되면 listup 폴더로 옮겨서 관리하려고 합니다.

**Reply by kwakjeeyoon** [reply]
저도 폴더로 관리하는 편이 더 편할것 같습니다!

**Reply by JiyouSeo** [reply]
네, 이해했습니다. 저도 찬성입니다.

**Comment by ppskj178** [comment]
해당 구조로 pr 올렸습니다.
#59
